src/dir.rs

```diff
@@ -161,7 +174,7 @@ impl PathDir {
     /// # Ok(()) } fn main() { try_main().unwrap() }
     /// ```
     pub fn join_abs<P: AsRef<Path>>(&self, path: P) -> Result<PathType> {
-        let joined = self.join(path.as_ref());
+        let joined = self.concat(path.as_ref())?;
```

Comment:
Technically this is a breaking change, since `join_abs()` now uses "concatenation" rather than "adjoinment". However, the use-case demonstrated in the doctest (concatenating a relative path) is unchanged.

What do you expect to happen if somebody takes `PathDir("/foo")` and calls `.join_abs("/bar")`, assuming both `/foo/bar` and `/bar` exist? Maybe `join_abs()` should stay as it was and there should be a new `concat_abs()`? Maybe the new `join_abs()` should be renamed to `concat_abs()`? Maybe the `PathOps` should be generic enough that the existing `concat()` method can be implemented for `PathDir`?

Reply:
do you think it makes sense to remove `join_abs`? I feel like it isn't really needed anymore.

Reply:
it can be done outside of this PR.

Reply:
I intended to go back and compare the inherent methods of all the `Path*` types with the trait methods and see what redundancies could be eliminated... but I kept changing the design of the traits to account for new problems I found, and I never got around to it. That work should probably still be done. Just looking at this particular example:

- `PathDir::join_abs()` takes a path, joins it to `self` and (if the path is relative and exists) returns a new `PathType'.
- `<PathDir as PathOps>::concat()` takes a path, joins it to `self` and (if the path is relative) returns a new `PathAbs`
- We could extend `PathType` with a new variant, `Missing(PathAbs)`, and teach our methods to return one of those in the case of a missing result, instead of an error.
- We could rename `PathDir::join_abs()` to `concat()`, so there'd only be one name to remember, although there would be two slightly different behaviours depending on which version you called.
- We could decide that the flexibility of two behaviours is not worth the cognitive overhead of having to choose between them, and just delete one.

That should probably be a followup ticket, though, you're right.